go/vt/vtgate/scatter_conn.go

```diff
@@ -166,6 +168,21 @@ func (stc *ScatterConn) ExecuteMultiShard(
 	var mu sync.Mutex
 	qr = new(sqltypes.Result)
 
+	if session.InLockSession() && session.TriggerLockHeartBeat() {
+		go func() {
```

Comment:
This is better than what we have now, but won't cover all use cases. For example, a bulk insert that takes longer than 5s will expire the lock.

I think the only reliable method would be to run an independent ticker that keeps sending heartbeats. But that is much more complicated than this. So, this is a good compromise.

Reply:
I will create a future task for it.